packages/test/test-end-to-end-tests/src/test/gc/gcStateResetInSummaries.spec.ts

```diff
@@ -0,0 +1,276 @@
+            [
+                [dataObjectFactory.type, Promise.resolve(dataObjectFactory)],
+            ],
+            undefined,
+            undefined,
+            { ...defaultRuntimeOptions, gcOptions: { gcAllowed, disableGC } },
+        );
+        return loadSummarizer(
+            provider, runtimeFactory, mainContainer.deltaManager.lastSequenceNumber, summaryVersion);
+    };
+
+    /**
+     * Callback that will be called by the document storage service whenever a summary is uploaded by the client.
+     * Update the summary context to include the summary proposal and ack handle as per the latest ack for the
+     * document.
+     */
+    function uploadSummaryCb(summaryTree: ISummaryTree, context: ISummaryContext): ISummaryContext {
+        latestUploadedSummary = summaryTree;
+        latestSummaryContext = context;
+        const newSummaryContext = { ...context };
+        // If we received an ack for this document, update the summary context with its information. The
+        // server rejects the summary if it doesn't have the proposal and ack handle of the previous
+        // summary.
+        if (latestAckedSummary !== undefined) {
+            newSummaryContext.ackHandle = latestAckedSummary.summaryAck.contents.handle;
+            newSummaryContext.proposalHandle = latestAckedSummary.summaryOp.contents.handle;
+        }
+        return newSummaryContext;
+    }
+
+    /** Generates a summary and returns the data store channel sub-tree */
+    async function getSummaryChannelsTree(
+        summarizerClient: { containerRuntime: ContainerRuntime, summaryCollection: SummaryCollection },
+    ) {
+        const summaryResult = await submitAndAckSummary(provider, summarizerClient, logger, true /* fullTree */);
+        latestAckedSummary = summaryResult.ackedSummary;
+
+        assert(
+            latestSummaryContext && latestSummaryContext.referenceSequenceNumber >= summaryResult.summarySequenceNumber,
+            `Did not get expected summary. Expected: ${summaryResult.summarySequenceNumber}. ` +
+            `Actual: ${latestSummaryContext?.referenceSequenceNumber}.`,
+        );
+        assert(latestUploadedSummary !== undefined, "Did not get a summary");
+
+        return (latestUploadedSummary.tree[channelsTreeName] as ISummaryTree)?.tree ?? latestUploadedSummary.tree;
+    }
+
+    /**
+     * Validates that GC ran by asserting that all data stores have GC state. Also, the data store whose id is
+     * unreferencedDataStoreId is marked as unreferenced as per the GC state.
+     */
+    async function validateGCRan(
+        summarizerClient: { containerRuntime: ContainerRuntime, summaryCollection: SummaryCollection },
+        unreferencedDataStoreId?: string,
+    ) {
+        const channelsTree = await getSummaryChannelsTree(summarizerClient);
+        for (const [ id, summaryObject ] of Object.entries(channelsTree)) {
+            if (summaryObject.type !== SummaryType.Tree) {
+                continue;
```

Comment:
should the test check to see if at least one object was processed?

Reply:
Good point. Also added additional check to validate we are processing data store entries.